aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -1321,22 +1469,30 @@ std::pair<Tensor, hidden_type> _cudnn_impl(
   Tensor hx, cx;
   std::tie(hx, cx) = unpack_hidden(hidden);
   int64_t hidden_size = hx.size(2);
-
+  // For LSTM models with projections hidden size could be different
+  if (cx.defined()) {
+    hidden_size = cx.size(2);
+  }
+  int64_t proj_size = hx.size(2);
```

Comment:
This is misleading for a case when there's no projection (instead of 0 it will be computed as `hx.size(2)`